stactools_landsat/stactools/landsat/utils.py

```diff
@@ -101,49 +102,51 @@ def transform_stac_to_stac(item: Item,
     del item.properties["eo:off_nadir"]
 
     if enable_proj:
-        try:
-            # If we can load the blue band, use it to add proj information
-            if item.assets.get("SR_B2.TIF"):
-                asset = item.assets["SR_B2.TIF"]
-            # SR_B10 is a fallback for SR_B2
-            elif item.assets.get("SR_B10.TIF"):
-                asset = item.assets["SR_B10.TIF"]
-            else:
-                raise ValueError('Asset SR_B2.TIF or SR_B2.TIF required')
-
-            with rasterio.open(asset.href) as opened_asset:
-                shape = [opened_asset.height, opened_asset.width]
-                transform = opened_asset.transform
-                crs = opened_asset.crs.to_epsg()
-
-            # Now we have the info, we can make the fields
-            item.ext.enable("projection")
-            item.ext.projection.epsg = crs
-
-            for name, asset in item.assets.items():
-                if asset.media_type == "image/vnd.stac.geotiff; cloud-optimized=true":
-                    item.ext.projection.set_transform(transform, asset=asset)
-                    item.ext.projection.set_shape(shape, asset=asset)
-                    asset.media_type = MediaType.COG
-
-        except RasterioIOError:
-            print("Failed to load blue band, so not handling proj fields")
+        # Enabled projection
+        item.ext.enable("projection")
+
+        shape = None
+        transform = None
+        crs = None
+        for name, asset in item.assets.items():
+            if "geotiff" in asset.media_type:
+                # retrieve shape, transform and crs from the first geotiff file among the assets
+                if not shape:
+                    try:
+                        with rasterio.open(asset.href) as opened_asset:
+                            shape = opened_asset.shape
+                            transform = opened_asset.transform
+                            crs = opened_asset.crs.to_epsg()
+                            # Check to ensure that all information is present
+                            if not shape or not transform or not crs:
+                                raise STACError(f"Failed setting shape, transform and csr from {asset.href}")
+
+                    except RasterioIOError as io_error:
+                        raise STACError(f"Failed loading geotiff, so not handling proj fields, {io_error}")
+
+                item.ext.projection.set_transform(transform, asset=asset)
+                item.ext.projection.set_shape(shape, asset=asset)
```

Comment:
This still looks like we're setting the transform and the shape N times, where N is the number of assets. I think this should be indented one more level to be inside of the `if not shape:` block.

```suggestion
                    item.ext.projection.set_transform(transform, asset=asset)
                    item.ext.projection.set_shape(shape, asset=asset)
```

Reply:
Morning Pete,
       in this case I'm not setting transform and shape to the item, I'm setting transform and shape to each asset and because of that it has to be inside of the loop.
       PYSATAC Library (projection.py):
                 asset.properties['proj:transform'] = transform => Line 425
                 asset.properties['proj:shape'] = shape => Line 379

Reply:
Hey @gadomski, the extension specifies that you can either set this at the root or per-asset.

We prefer it per-asset, and that's how the Element-84 S-2 COGs implemented it, see here: https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l2a-cogs/items

So we do want to do this in the loop for each asset.